### PR TITLE
feat: Ball lightning trails

### PR DESCRIPTION
**Pull request by scarf005**
Dungeon Crawl Stone Soup 0 30-a0-341-g68270651_01 
Dungeon Crawl Stone Soup 0 30-a0-341-g68270651_02

make ball lightnings leave harmless trail (similar to foxfire trail), because it's pretty.

### CODE REVIEW

File: crawl-ref/source/cloud-type.h [hunk 1]
```diff
@@ -42,6 +42,7 @@ enum cloud_type
     CLOUD_EMBERS,
 #endif
     CLOUD_FLAME,
+    CLOUD_ELECTRICITY,
```
Comment (PleasingFungus) [inline crawl-ref/source/cloud-type.h:45]: Inserting an enum into the middle of a list of enums will break save compatibility if the enum is marshalled (included in save files), as this one is. Anyone who loads up an old save after this change will find all degeneration clouds have turned into electricity clouds, and all blastsparks turned into degeneration.

You need to insert this at the end of the enum, right before NUM_CLOUD_TYPES.

File: crawl-ref/source/cloud.cc [hunk 2]
```diff
@@ -302,6 +302,11 @@ static const cloud_data clouds[] = {
       ETC_FIRE,                                   // colour
       { TILE_CLOUD_FLAME, CTVARY_RANDOM },        // tile
     },
+    // CLOUD_ELECTRICITY,
+    { "wisps of electric spark", nullptr,         // terse, verbose name
```
Comment (PleasingFungus) [inline crawl-ref/source/cloud.cc:306]: "wisps of electric spark" isn't quite valid English. I'd recommend just 'sparks'.

File: crawl-ref/source/cloud.cc [hunk 3]
```diff
@@ -302,6 +302,11 @@ static const cloud_data clouds[] = {
       ETC_FIRE,                                   // colour
       { TILE_CLOUD_FLAME, CTVARY_RANDOM },        // tile
     },
+    // CLOUD_ELECTRICITY,
```
Comment (PleasingFungus) [inline crawl-ref/source/cloud.cc:305]: The new cloud needs an entry in `dat/descript/clouds.txt`.

**Review by PleasingFungus — CHANGES_REQUESTED** [review]
This is very cool! Many apologies for the long delay in review.

**Comment by scarf005** [comment]
closing in order to reopen PR (#2925) from another fork.

**Pull request closed without merging**